src/index/inverted_index.rs

```diff
@@ -93,8 +93,14 @@ impl IndexBuilder {
         self.terms.push(term.to_string());
     }
 
-    pub fn insert_document(&mut self, name: &str) {
+    pub fn push_posting(&mut self, term_id: u32, doc_id: u32, tf: u32) {
+        self.posting_lists[term_id as usize].push((doc_id, tf));
```

Comment:
Since doc_id is incremental we are guaranteed to have increasing order, right?

Reply:
Yes, but it's worth a comment, since the function name doesn't imply this. I'll add